### PR TITLE
Update Hoxy, add HTTPS support

### PR DESCRIPTION
**Pull request by mitchhentges**
Also fixes tests to match new hoxy API

Currently, will enable HTTPS part of proxy only if "root-ca" cert and key files available in current directory. There's no fancy UI or anything, but hey, it's progress :)

### CODE REVIEW

**Comment by mitchhentges** [comment]
@nerdbeere how often does the team generally check PRs?

**Comment by nerdbeere** [comment]
Wow, thanks. I will look into it.
Sorry I was on vacation for the last weeks. Normally I look on a daily basis for PRs.

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,5 @@ build
 package
 node_modules
 binaries
+.idea/
```
Comment (nerdbeere) [inline .gitignore]: Could you please add this to your global `.gitignore` and remove it here?

Comment (mitchhentges) [inline .gitignore]: Fixed!

**Comment by nerdbeere** [comment]
Again thanks for your contribution!
Would be super awesome if you could provide a little documentation about how you can use this feature.

**Comment by mitchhentges** [comment]
You bet, I edited the James Github wiki

**Pull request merged**